src/System.Runtime.Serialization.Formatters/tests/FormatterServicesTests.cs

```diff
@@ -29,21 +34,260 @@ public void GetSerializableMembers_Interface()
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_OpenGenericClass_TestData))]
+        public void GetUninitializedObject_OpenGenericClass_ThrowsMemberAccessException(Type type)
+        {
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<MemberAccessException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        public interface IGenericClass
+        {
+            int Value { get; set; }
+        }
+
+        public class GenericClass<T> : IGenericClass
+        {
+            public int Value { get; set; }
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_TestData()
+        {
+            yield return new object[] { typeof(ArgIterator) };
+            yield return new object[] { typeof(RuntimeArgumentHandle) };
+            yield return new object[] { typeof(TypedReference) };
+
+            yield return new object[] { typeof(Span<int>) };
+            yield return new object[] { typeof(ReadOnlySpan<int>) };
+        }
+
+        public static IEnumerable<object[]> GetUninitializedObject_ByRefLikeType_NetCore_TestData()
+        {
+            yield return new object[] { Type.GetType("System.ByReference`1[System.Int32]") };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_TestData))]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_NetCore_TestData))]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The full .NET framework supports GetUninitializedObject for by ref like types")]
+        public void GetUninitializedObject_ByRefLikeType_NetCore_ThrowsNotSupportedException(Type type)
+        {
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetUninitializedObject(type));
+            Assert.Throws<NotSupportedException>(() => FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Theory]
+        [MemberData(nameof(GetUninitializedObject_ByRefLikeType_TestData))]
+        [SkipOnTargetFramework(~TargetFrameworkMonikers.NetFramework, "The coreclr doesn't support GetUninitializedObject for by ref like types")]
+        public void GetUninitializedObject_ByRefLikeType_Netfx_ThrowsNotSupportedException(Type type)
+        {
+            Assert.NotNull(FormatterServices.GetUninitializedObject(type));
+            Assert.NotNull(FormatterServices.GetSafeUninitializedObject(type));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "The full .NET framework supports GetUninitializedObject for shared generic instances")]
+        public void GetUniniitalizedObject_SharedGenericInstance_NetCore_ThrowsNotSupportedException()
+        {
+            Type canonType = Type.GetType("System.__Canon");
```

Comment:
I think this is an important test, as it exercises a direct codepath in the coreclr implementation of GetUninitializedObject. And, it calls out a difference between netfx and netcoreapp, whereby GetSafeUninitializedObject doesn't throw in netfx, but does in netcoreapp due to runtime differences.

What I have done, however, is changed the condition for the skipping to be `~TargetFrameworkMonikers.Netcoreapp` to avoid errors for uataop and mono